pyfstat/grid_based_searches.py

```diff
@@ -1161,6 +1234,248 @@ def _get_savetxt_fmt_dict(self):
         return fmt_dict
 
 
+class SearchOverGridFile(TransientGridSearch):
+    """A search over a grid defined in an input text file.
+
+    Such a file can be generated for example with
+    lalapps_ComputeFstatistic_v2 using the --outputGrid option.
+
+    WARNING: The header file of the grid file is NOT yet parsed
+    for consistency of CFSv2 options (such as reference time)
+    with the PyFstat options,
+    so please ensure yourself that these match!
+
+    Most parameters are the same as for `GridSearch`, `TransientGridSearch`
+    and the `core.ComputeFstat` class,
+    only the additional ones are documented here.
+
+    NOTE: `nsegs>1` is incompatible with transient options.
+    """
+
+    @utils.initializer
+    def __init__(
+        self,
+        label,
+        outdir,
+        sftfilepattern,
+        gridfile,
+        tref,
+        minStartTime=None,
+        maxStartTime=None,
+        nsegs=1,
+        BSGL=False,
+        BtSG=False,
+        minCoverFreq=None,
+        maxCoverFreq=None,
+        detectors=None,
+        SSBprec=None,
+        RngMedWindow=None,
+        injectSources=None,
+        input_arrays=False,
+        assumeSqrtSX=None,
+        transientWindowType=None,
+        t0Band=None,
+        tauBand=None,
+        tauMin=None,
+        dt0=None,
+        dtau=None,
+        outputTransientFstatMap=False,
+        outputAtoms=False,
+        tCWFstatMapVersion="lal",
+        cudaDeviceName=None,
+        earth_ephem=None,
+        sun_ephem=None,
+        clean=False,
+        reading_method="numpy",
+    ):
+        """
+        Parameters
+        ----------
+        gridfile: str
+            filename of the grid file to load
+        """
+
+        self._set_init_params_dict(locals())
+        if self.nsegs > 1 and self.transientWindowType is not None:  # pragma: no cover
+            raise ValueError(
+                f"nsegs={self.nsegs} is incompatible with transient options (transientWindowType={self.transientWindowType})."
+            )
+        os.makedirs(outdir, exist_ok=True)
+        self.set_out_file()
+        if reading_method == "numpy":
+            self._read_grid_with_numpy()
+        elif reading_method == "pandas":
+            self._read_grid_with_pandas()
+        else:  # pragma: no cover
+            raise ValueError(
+                f"Invalid reading method: {reading_method}. Expected 'numpy' or 'pandas'."
+            )
+        self.search_keys = list(self.grid.dtype.names)  # this is okey with pandas
```

Comment:
```suggestion
        self.search_keys = list(self.grid.dtype.names)
```
Comment no longer needed, as we convert the pandas dataframe back to numpy already.